icepyx/quest/dataset_scripts/argo.py

```diff
@@ -0,0 +1,478 @@
+import numpy as np
+import pandas as pd
+import requests
+
+from icepyx.core.spatial import geodataframe
+from icepyx.quest.dataset_scripts.dataset import DataSet
+
+
+class Argo(DataSet):
+    """
+    Initialises an Argo Dataset object
+    Used to query physical and BGC Argo profiles
+
+    Examples
+    --------
+    # example with profiles available
+    >>> reg_a = Argo([-154, 30,-143, 37], ['2022-04-12', '2022-04-26'])
+    >>> reg_a.search_data()
+    >>> print(reg_a.profiles[['pres', 'temp', 'lat', 'lon']].head())
+        pres    temp     lat      lon
+    0   3.9  18.608  33.401 -153.913
+    1   5.7  18.598  33.401 -153.913
+    2   7.7  18.588  33.401 -153.913
+    3   9.7  18.462  33.401 -153.913
+    4  11.7  18.378  33.401 -153.913
+
+    # example with no profiles
+    >>> reg_a = Argo([-55, 68, -48, 71], ['2019-02-20', '2019-02-28'])
+    >>> reg_a.search_data()
+    Warning: Query returned no profiles
+    Please try different search parameters
+
+    See Also
+    --------
+    DataSet
+    GenQuery
+    """
+
+    # Note: it looks like ArgoVis now accepts polygons, not just bounding boxes
+    def __init__(self, aoi, toi, params=["temperature"], presRange=None):
+        # super().__init__(boundingbox, timeframe)
+        self.params = self._validate_parameters(params)
+        self.presRange = presRange
+        self._spatial = aoi
+        self._temporal = toi
+        # todo: verify that this will only work with a bounding box (I think our code can accept arbitrary polygons)
+        assert self._spatial._ext_type == "bounding_box"
```

Comment:
@RomiP The tests fail on this assertion. It's because in the test suite we create an Argo object directly, without going through Quest. This seems logical from a testing perspective, BUT, because (1) Argo is a subclass (is it? or is it something else since it never directly inherits anything beyond the shell/template?) of DataSet (which is a template), and (2) Quest is (independently) a subclass of SuperQuery and where the `super().__init__` (spatial and temporal init) action happens, the `__init__` of Argo doesn't actually call SuperQuery, so the `_spatial` attribute isn't there and the assertion fails. So my questions are: is having an object that cannot be initialized directly bad practice? If not, that means we should only ever create/access Argo objects via Quest, so we should somehow have that documented/indicated and presumably also create the Argo objects for testing via Quest.